rust/sbp/src/sbp_string.rs

```diff
@@ -37,6 +38,94 @@ where
     }
 }
 
+impl<T: AsRef<[u8]>> SbpString<T, Unterminated> {
+    /// Checked unterminated SbpString builder, verify last byte is non null
+    pub fn unterminated(data: T) -> Result<Self, UnterminatedError> {
+        match data.as_ref().last() {
+            Some(l) if l != &0 => Ok(Self::new(data)),
```

Comment:
I think an unterminated string ending in a null byte is actually okay? @silverjam do you know?

Reply:
I don't think we enforce this anywhere in firmware so this really depends on intended usage-- if we expect to use this construct unterminated SbpStrings from a remote device then I think this will fail because some areas where a string is expected to be "unterminated" we'll end up inserting a null terminated one just because that's the convention in C.

Reply:
hmmm should we skip unterminated then 🤔